components/camel-jpa/src/main/java/org/apache/camel/processor/idempotent/jpa/JpaMessageIdRepository.java

```diff
@@ -130,8 +125,8 @@ public Boolean doInTransaction(TransactionStatus status) {
             }
         });
 
-        LOG.debug("add {} -> {}", messageId, rc);
-        return rc;
+        LOG.debug("add {} -> {}", messageId, rc[0]);
```

Comment:
Maybe it would be a good opportunity to improve this log message. It's not very clear what information it is trying to convey.